### PR TITLE
Python: Use Prettier to format TOML and more files

### PR DESCRIPTION
**Pull request by deepyaman**
Came across https://github.com/apache/iceberg/pull/6745#discussion_r1100602879 while looking into something else, and figured I could do it quickly. :)

Prettier is a well-established auto-formatter for many types of files. I've personally used it on a lot of projects int he past. The TOML integration is community-built (but linked to from the official Prettier website and docs) and no longer maintained, but still seems to function fine (including in testing on the case from the aforementioned comment). On the whole, it did seem to help standardize formatting of the `pyproject.toml` file.

### CODE REVIEW

File: python/mkdocs/docs/api.md [hunk 1]
```diff
@@ -307,15 +307,18 @@ scan = table.scan(
...
 ```
 
 In this case it is up to the engine itself to filter the file itself. Below, `to_arrow()` and `to_duckdb()` that already do this for you.
 
 ### Apache Arrow
 
 !!! note "Requirements"
-    This requires [PyArrow to be installed](index.md)
+
+This requires [PyArrow to be installed](index.md)
```
Comment (Fokko) [inline python/mkdocs/docs/api.md:321]: This breaks the note unfortunately.

File: python/mkdocs/docs/feature-support.md [hunk 2]
```diff
@@ -21,21 +21,21 @@ The goal is that the python library will provide a functional, performant subset
...
-| Update Current Snapshot  |    X  |        |
-| Create Table             |    X  |   X    |
-| Rename Table             |    X  |   X    |
-| Drop Table               |    X  |   X    |
-| Alter Table              |    X  |        |
-| Set Table Properties     |    X  |        |
-| Create Namespace         |    X  |   X    |
-| Drop Namespace           |    X  |   X    |
-| Set Namespace Properties |    X  |   X    |
+| Operation                | Java | Python |
```
Comment (Fokko) [inline python/mkdocs/docs/feature-support.md:24]: I checked and this looks good: 
image

**Review by Fokko — COMMENTED** [review]
Thanks @deepyaman, this looks great! I have one comment about the note, would it be possible to have to linker accept that? Otherwise, it breaks the note.

File: python/mkdocs/docs/api.md (hunk 1 above)
Comment (Fokko) [reply python/mkdocs/docs/api.md:321]: image

😭

Comment (deepyaman) [reply python/mkdocs/docs/api.md:321]: Hmm... I haven't used `mkdocs` previously, but it seems like their admonition syntax isn't supported by Prettier (or the underlying parser). There are a number of existing issues (e.g. https://github.com/prettier/prettier/issues/3362 and https://github.com/prettier/prettier/issues/12985) highlighting this, and the recommendation seems to wrap them in `<!-- prettier-ignore-start -->`, `<!-- prettier-ignore-end -->`. I can do that if it's acceptable?

Comment (Fokko) [reply python/mkdocs/docs/api.md:321]: I'm aware of the note syntax being a bit odd indeed, other linters are also having issues with that. I'm okay with the ignore comments 👍🏻

**Review by Fokko — APPROVED** [review]
@deepyaman Love it, thanks for picking this up!

**Pull request merged**